src/server/scripts/Zandalar/AtalDazar/boss_rezan.cpp

```diff
@@ -0,0 +1,350 @@
+    { -793.8f,   2239.09f, 641.832f },
+    { -859.222f, 2256.25f, 643.136f },
+    { -871.375f, 2258.77f, 643.438f },
+    { -804.901f, 2254.52f, 641.964f },
+    { -884.057f, 2257.15f, 642.643f },
+    { -818.212f, 2259.57f, 642.465f },
+    { -830.214f, 2265.17f, 642.486f },
+    { -897.104f, 2259.22f, 641.901f },
+    { -874.649f, 2273.9f,  642.642f },
+    { -809.342f, 2272.22f, 641.905f },
+    { -821.203f, 2279.28f, 642.488f },
+    { -893.123f, 2276.57f, 642.583f },
+    { -848.333f, 2288.46f, 642.486f },
+    { -797.328f, 2282.58f, 641.182f },
+    { -900.995f, 2285.5f,  641.773f },
+    { -804.385f, 2293.2f,  641.488f },
+    { -895.255f, 2298.0f,  642.486f },
+    { -880.141f, 2301.81f, 642.501f },
+    { -826.514f, 2306.46f, 642.699f },
+    { -871.58f,  2311.03f, 642.487f },
+    { -803.264f, 2310.29f, 641.707f },
+    { -817.41f,  2313.88f, 641.638f },
+    { -889.821f, 2315.4f,  642.486f },
+    { -834.929f, 2321.18f, 643.028f },
+    { -861.599f, 2322.33f, 642.5f   },
+    { -872.771f, 2324.00f, 642.77f  },
+    { -825.752f, 2329.17f, 642.247f },
+    { -810.163f, 2327.19f, 641.401f },
+    { -884.835f, 2331.72f, 642.487f },
+    { -843.49f,  2335.39f, 642.765f },
+    { -895.611f, 2334.97f, 642.425f },
+    { -874.451f, 2340.77f, 643.405f },
+    { -817.785f, 2342.43f, 641.914f },
+    { -861.078f, 2344.69f, 642.685f },
+    { -838.155f, 2345.16f, 642.733f },
+    { -869.793f, 2352.61f, 642.515f }
+};
+
+// 122963 - Rezan
+struct boss_rezan : public BossAI
+{
+    boss_rezan(Creature* creature) : BossAI(creature, DATA_REZAN)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        for (Position spawnPoint : PilesOfBonesPosition)
+        {
+            me->CastSpell(spawnPoint, SPELL_PILE_OF_BONES_AT_SPAWN, true);
+            me->CastSpell(spawnPoint, SPELL_PILE_OF_BONES_AT_SLOW, true);
+        }
+    }
+
+    void PassengerBoarded(Unit* who, int8 /*seatId*/, bool apply) override
+    {
+        if (apply && who->ToPlayer())
+            DoCast(who, SPELL_DEVOUR, TRIGGERED_IGNORE_CAST_IN_PROGRESS);
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SetBossState(DATA_REZAN, FAIL);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+
+        summons.DespawnAll();
+
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        instance->SetBossState(DATA_REZAN, IN_PROGRESS);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me);
+
+        events.ScheduleEvent(EVENT_TERRIFYING_VISAGE, 12200ms);
+        events.ScheduleEvent(EVENT_TAIL, 6100ms);
+        events.ScheduleEvent(EVENT_SERRATED_TEETH, 8500ms);
+        events.ScheduleEvent(EVENT_PURSUIT, 21900ms);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        instance->SetBossState(DATA_REZAN, DONE);
```

Comment:
unnecessary as _JustDied() does exactly this 
![image](https://github.com/user-attachments/assets/f370bc33-456c-477e-a5a2-47b9680b6b00)